### PR TITLE
[JSON-API] Add query store metrics

### PR DESCRIPTION
**Pull request by realvictorprm**
Adds the metrics introduced in #11769 to the 1.11.x release branch to main.

CHANGELOG_BEGIN

- [JSON-API] added metrics to separately track:
    - time taken to update query-store ACS (from ledger)
    - lookup times for the query store

CHANGELOG_END

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by stefanobaghino-da — COMMENTED** [review]
LGTM

**Comment by realvictorprm** [comment]
Ensured once locally via our tests that the new metrics are spit out and everything works :)

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/ContractsService.scala
Comment (S11001001) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/ContractsService.scala:399]: Side effects are dangerous.

```suggestion
            it: => doobie.ConnectionIO[A],
        ): doobie.ConnectionIO[A] = {
          for {
            _ <- fconn.pure(())
            ctx <- fconn.pure(timer.time())
            res <- it
            _ <- fconn.pure(ctx.stop())
```

Comment (victormueller-da) [reply ledger-service/http-json/src/main/scala/com/digitalasset/http/ContractsService.scala:399]: Good point! I assume that the argument is not lazy for pure then? Nevertheless in the end it shouldn't be seen as pure at all.

Comment (S11001001) [reply ledger-service/http-json/src/main/scala/com/digitalasset/http/ContractsService.scala:399]: Indeed, it is not pure.

Comment (S11001001) [reply ledger-service/http-json/src/main/scala/com/digitalasset/http/ContractsService.scala:399]: I'm honestly not 100% sure whether this is good enough or future-proof for the timer start even now. Flatmapped pures can be fused. Given the calling context, this is probably fine, though.

Putting something on the right of a flatmap rather than of a map, when the left side of the flatmap is not a pure, is always a good way to delay it though. So the timer stop is just fine with this change.

**Review by S11001001 — APPROVED** [review]

**Comment by stefanobaghino-da** [comment]
@cocreature Possible flake in `//compiler/lsp-tests:lsp-tests` that you may be aware of. It occurred in the second attempt in this [build](https://dev.azure.com/digitalasset/daml/_build/results?buildId=95789&view=logs&j=50a69d0a-7972-5459-cdae-135ee6ebe312&t=f7e418c2-3897-54b7-7549-71d829771cba). The log excerpt for that test is at least ~2K lines long and I don't know what the relevant bits are, so I'll skip it and leave it to you to look at the logs.

**Comment by stefanobaghino-da** [comment]
@realvictorprm This PR failed to build due to a flake. I re-started the build. Any reason to hold back from merging?

**Comment by cocreature** [comment]
> @cocreature Possible flake in `//compiler/lsp-tests:lsp-tests` that you may be aware of. It occurred in the second attempt in this [build](https://dev.azure.com/digitalasset/daml/_build/results?buildId=95789&view=logs&j=50a69d0a-7972-5459-cdae-135ee6ebe312&t=f7e418c2-3897-54b7-7549-71d829771cba). The log excerpt for that test is at least ~2K lines long and I don't know what the relevant bits are, so I'll skip it and leave it to you to look at the logs.

thank you, tracked in https://github.com/digital-asset/daml/issues/11873. If you hit it again please let us know and we can mark it flaky until we have a fix.

**Comment by victormueller-da** [comment]
> @realvictorprm This PR failed to build due to a flake. I re-started the build. Any reason to hold back from merging?

Just waiting for the build to pass, then it should be auto merged.

**Comment by victormueller-da** [comment]
PS: Thanks for re running the build! :)

**Comment by stefanobaghino-da** [comment]
@realvictorprm Get ready to start it again, I'm looking at the logs and it looks there's another failure.

**Pull request merged**
